### PR TITLE
[FLINK-17612][python][sql-client] add the parse logic of Python command line options to CliOptionsParser in SQL Client.

### PR DESCRIPTION
**Pull request by WeiZhong94**
## What is the purpose of the change

*This pull request adds the parse logic of Python command line options to CliOptionsParser in SQL Client.*

## Brief change log

  - *Add the parse logic of Python command line options to CliOptionsParser in SQL Client.*
  - *Append the Python configurations from CliOptions to Session Environment.*

## Verifying this change

This change can be verified as follows:

  - *Start a local standalone flink cluster.*
  - *Prepare a sql client session environment file with python UDF declaration.*
  - *Added test that validates that TaskInfo is transferred only once across recoveries*
  - *Launch the sql client with command line option "-pyfs xxx.py", "xxx.py" is the path of the python file that defines the python UDF.*
  - *The sql client should up without any exception.*

**NOTE:** After adding the PyFlink e2e test, the changes in this PR will be automatically verified.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliOptionsParser.java [hunk 1]
```diff
@@ -331,4 +354,18 @@ private static String checkSessionId(CommandLine line) {
...
+		try {
+			Class<?> clazz = Class.forName(
+				"org.apache.flink.python.util.PythonDependencyUtils",
+				true,
+				Thread.currentThread().getContextClassLoader());
+			Method parsePythonDependencyConfiguration =
+				clazz.getMethod("parsePythonDependencyConfiguration", CommandLine.class);
+			return (Configuration) parsePythonDependencyConfiguration.invoke(null, line);
+		} catch (ClassNotFoundException | NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+			throw new SqlClientException("Parse the Python command line options failed.", e);
```
Comment (dianfu) [inline flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliOptionsParser.java:368]: Failed to parse the Python command line options

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java [hunk 2]
```diff
@@ -90,6 +96,7 @@ private void start() {
 
 			// create CLI client with session environment
 			final Environment sessionEnv = readSessionEnvironment(options.getEnvironment());
+			appendPythonConfig(sessionEnv, options.getPythonConfiguration());
```
Comment (dianfu) [inline flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java:99]: could use the `enrich` method defined in Environment as following:
`
final Environment sessionEnvWithPythonConfiguration = Environment.enrich(
				sessionEnv, options.getPythonConfiguration().toMap(), Collections.emptyMap());
`

Comment (WeiZhong94) [reply flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java:99]: It seems the `enrich()` method of the `ConfigurationEntry` only accepts the configurations that start with "table.". As the python configurations all start with "python.", the `enrich` method does not work for python configurations.

**Comment by WeiZhong94** [comment]
@dianfu Thanks for your review! I have updated this PR.

**Pull request merged**